Tasks/AzureRmWebAppDeploymentV4/operations/KuduServiceUtility.ts

```diff
@@ -176,7 +177,47 @@ export class KuduServiceUtility {
             var deploymentMessage = this._getUpdateHistoryRequest(true, null, customMessage).message;
             queryParameters.push('message=' + encodeURIComponent(deploymentMessage));
 
-            let deploymentDetails = await this._appServiceKuduService.zipDeploy(packagePath, queryParameters);
+            let deploymentDetails = await this._appServiceKuduService.zipDeploy(packagePath, queryParameters, addChecksumHeader);
+           
+            await this._processDeploymentResponse(deploymentDetails);
+            if(appOffline) {
+                await this._appOfflineKuduService(physicalRootPath, false);
+            }
+           
+            console.log(tl.loc('PackageDeploymentSuccess'));
+            return deploymentDetails.id;
+        }
+        catch(error) {
+            tl.error(tl.loc('PackageDeploymentFailed'));
+            throw Error(error);
+        }
+    }
+
+    public async deployUsingOneDeploy(packagePath: string, isClean: boolean, appOffline?: boolean, customMessage?: any, packageType?:string, addChecksumHeader?: boolean): Promise<string> {
+        try {
+            console.log(tl.loc('PackageDeploymentInitiatedWithOneDeploy'));
+
+            if(appOffline) {
+                await this._appOfflineKuduService(physicalRootPath, true);
+                tl.debug('Wait for 5 seconds for app_offline to take effect');
+                await webClient.sleepFor(5);
+            }
+
+            if (!packageType){
+                packageType = 'Zip'
```

Comment:
IMO we need debug log line here - "no packageType: fallback to zip package type"